modules/library/referencing/src/main/java/org/geotools/referencing/wkt/DefaultUnitParser.java

```diff
@@ -32,34 +35,28 @@
 public class DefaultUnitParser extends BaseGT2Format {
 
     private static final DefaultUnitParser UNITPARSER = new DefaultUnitParser();
-    protected HashMap<UnitWrapper, Unit<?>> unitWrapperToUnitMap = new HashMap<>();
-
-    //    /**
-    //     * Gets a UnitFormat configured to parse units. Since usually we don't know the citation
-    // in use
-    //     * for a particular unit literal definition, this format includes the aliases of EPSG and
-    // ESRI
-    //     * citations, in order to be able to parse the widest possible range of units.
-    //     *
-    //     */
-    //    public static DefaultUnitParser getInstance() {
-    //        return UNITPARSER;
-    //    }
-
-    public static DefaultUnitParser getInstance(
-            Flavor flavour) { // to avoid confusions with parent class
+
+    protected HashMap<UnitWrapper, Unit<?>> unitWrapperToUnitMap;
+
+    public static DefaultUnitParser getInstance() {
+        return UNITPARSER;
+    }
+
+    public static DefaultUnitParser getInstance(Flavor flavour) {
         return UNITPARSER;
     }
 
     DefaultUnitParser() {
-        initUnits(Units.getDefaultFormat());
         esriLabelsAndAliases(this);
         // add epsg labels the latest, to override esri ones if they collide
         epsgLabelsAndAliases(this);
     }
 
     @Override
     protected void addUnit(Unit<?> unit) {
+        if (unitWrapperToUnitMap == null) {
+            unitWrapperToUnitMap = new HashMap<>();
+        }
```

Comment:
need to check here, as we call this from a parent constructors, which means the field wouldn't be initialized yet.